lib/AST/ASTContext.cpp

```diff
@@ -908,6 +914,75 @@ FuncDecl *ASTContext::getEqualIntDecl() const {
   return nullptr;
 }
 
+FuncDecl *ASTContext::getMutatingXorIntDecl() const {
```

Comment:
I've just pushed a new commit that factors out the common code for this and for the intrinsic library function lookups.